test/c/test_split_2q_unitaries.c

```diff
@@ -0,0 +1,192 @@
+#include <string.h>
+
+int test_split_2q_unitaries_no_unitaries(void) {
+    QkCircuit *qc = qk_circuit_new(5, 0);
+    for (uint32_t i = 0; i < qk_circuit_num_qubits(qc) - 1; i++) {
+        uint32_t qargs[2] = {i, i + 1};
+        for (uint32_t j = 0; j < i + 1; j++) {
+            qk_circuit_gate(qc, QkGate_CX, qargs, NULL);
+        }
+    }
+    QkTranspileLayout *split_result =
+        qk_transpiler_pass_standalone_split_2q_unitaries(qc, 1 - 1e-16, true);
+    int result = Ok;
+    if (split_result != NULL) {
+        result = EqualityError;
+        printf("Permutation returned for a circuit that shouldn't split\n");
+        goto cleanup;
+    }
+    QkOpCounts counts = qk_circuit_count_ops(qc);
+    if (counts.len != 1) {
+        printf("More than 1 type of gate in the circuit\n");
+        result = EqualityError;
+        goto ops_cleanup;
+    }
+    for (int i = 0; i < counts.len; i++) {
+        int gate = strcmp(counts.data[i].name, "cx");
+        if (gate != 0) {
+            printf("gates changed when there should be no circuit changes\n");
+            result = EqualityError;
+            goto cleanup;
+        }
+    }
+ops_cleanup:
+    qk_opcounts_free(counts);
+cleanup:
+    qk_circuit_free(qc);
+    return result;
+}
+
+int test_split_2q_unitaries_x_y_unitary(void) {
+    QkCircuit *qc = qk_circuit_new(2, 0);
+    QkComplex64 c0 = {0., 0.};
+    QkComplex64 neg_im = {0., -1.};
+    QkComplex64 im = {0., 1.};
+    QkComplex64 unitary[16] = {
+        c0, c0, c0, neg_im, c0, c0, neg_im, c0, c0, im, c0, c0, im, c0, c0, c0,
+    };
+    uint32_t qargs[2] = {0, 1};
+    qk_circuit_unitary(qc, unitary, qargs, 2, true);
+    QkTranspileLayout *split_result =
+        qk_transpiler_pass_standalone_split_2q_unitaries(qc, 1 - 1e-16, true);
+    int result = Ok;
+    if (split_result != NULL) {
+        result = EqualityError;
+        printf("Permutation returned for a circuit that shouldn't isn't a swap equivalent\n");
+        goto cleanup;
+    }
+    QkOpCounts counts = qk_circuit_count_ops(qc);
+    if (counts.len != 1) {
+        printf("More than 1 type of gate in the circuit\n");
+        result = EqualityError;
+        goto ops_cleanup;
+    }
+    for (int i = 0; i < counts.len; i++) {
+        int gate = strcmp(counts.data[i].name, "unitary");
+        if (gate != 0) {
+            printf("Gates outside expected set in output circuit\n");
+            result = EqualityError;
+            goto ops_cleanup;
+        }
+        unsigned int count = counts.data[i].count;
+        if (count != 2) {
+            printf("Unexpected gate counts found\n");
+            result = EqualityError;
+            goto ops_cleanup;
+        }
+    }
+    QkCircuitInstruction inst;
+    for (size_t i = 0; i < qk_circuit_num_instructions(qc); i++) {
+        qk_circuit_get_instruction(qc, i, &inst);
+        if (inst.num_qubits != 1) {
+            printf("Gate %d operates on more than 1 qubit: %zu\n", i, inst.num_qubits);
```

Comment:
```suggestion
            printf("Gate %zu operates on more than 1 qubit: %u\n", i, inst.num_qubits);
```
Compiler screams: _use the right types!_